### PR TITLE
feat(lang): add support for helm chart templates

### LINKED ISSUE #2251: feature: Improve support for Helm templates
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

When editing a [Helm](https://helm.sh/) template with `extras.lang.yaml` enabled, there are many issues due to `yamlls` running even though the files should be identified as `helm` (or `gotmpl`). This is because even though the templates have a `.yaml` file extension, they are actually go templates for generating yaml.

image

### Describe the solution you'd like

Minimally, some configuration to prevent `yamlls` from attaching to these files is an okay solution.

Ideally, some form of support for `helm-ls` for these "yaml" files would be great, with some syntax highlighting (I apologize if I don't fully understand and these things are unrelated).

### Describe alternatives you've considered

I found [this](https://github.com/mrjosh/helm-ls#neovim-using-nvim-lspconfig) documentation which works, but had to be done manually.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pbnj**
Closes #2251 

See related discussion #1389

### CODE REVIEW

File: lua/lazyvim/plugins/extras/lang/helm.lua [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require("lazyvim.util.lsp").on_attach(function(client, buffer)
+  if client.name == "docker_compose_language_service" then
+    if vim.api.nvim_buf_get_option(buffer, "filetype") == "helm" then
```
Comment (siraphobk) [inline lua/lazyvim/plugins/extras/lang/helm.lua:3]: The `require("lazyvim.util.lsp").on_attach` is deprecated. I think it's better to use this instead.

```lua
require("lazyvim.util.lsp").lsp.on_attach
```

Also, the `vim.api.nvim_buf_get_option` is deprecated. My take is to use this.

```lua
vim.api.nvim_get_option_value("filetype", { buf = buffer })
```

**Comment by siraphobk** [comment]
Anyway, I wonder if you're facing the same problem as me. My helm template still shows the inline errors despite having vim-helm setup.

image

Here's my LspInfo.

``` 
 Language client log: /home/me/.local/state/nvim/lsp.log
 Detected filetype:   helm
 
 2 client(s) attached to this buffer: 
 
 Client: yamlls (id: 1, bufnr: [3])
 	filetypes:       yaml, yaml.docker-compose
 	autostart:       true
 	root directory:  /home/me/Workspaces/m/development/helm-charts
 	cmd:             /home/me/.local/share/nvim/mason/bin/yaml-language-server --stdio
 
 Client: helm_ls (id: 2, bufnr: [3])
 	filetypes:       helm
 	autostart:       true
 	root directory:  /home/me/Workspaces/m/development/helm-charts/charts/naaraan
 	cmd:             /home/me/.local/share/nvim/mason/bin/helm_ls serve
 
 Configured servers list: lua_ls, tailwindcss, yamlls, tsserver, pyright, gopls, ruff_lsp, dockerls, graphql, docker_compose_language_service, rust_analyzer, eslint, taplo, marksman, helm_ls, jsonls
```

Do you have any ideas about how to fix this?

File: lua/lazyvim/plugins/extras/lang/helm.lua (hunk 1 above)
Comment (pbnj) [reply lua/lazyvim/plugins/extras/lang/helm.lua:3]: When I use `require("lazyvim.util.lsp").lsp.on_attach`, I get an error 
```
undefined field `lsp`
```

Did you mean `require("lazyvim.util").lsp.on_attach`?
Isn't this the same as `require("lazyvim.util.lsp").on_attach`?

**Comment by pbnj** [comment]
@siraphobk  - should be fixed now.

File: lua/lazyvim/plugins/extras/lang/helm.lua (hunk 1 above)
Comment (siraphobk) [reply lua/lazyvim/plugins/extras/lang/helm.lua]: Whoops, my bad. You are certainly correct.

**Comment by siraphobk** [comment]
@pbnj Thanks, man.🎉 I'll let you know if there're any other issues.

**Comment by ingcrengifo** [comment]
when will this pr be mixed?

**Comment by folke** [comment]
Thanks!

**Pull request merged**